### PR TITLE
refactor: account select & sendform components

### LINKED ISSUE #902: 【MantaPay v4】Need to show two icons when two wallets are connecting

### LINKED ISSUE #907: 【MantaPay v4】Can't detect Talisman account

### PR DESCRIPTION
**Pull request by jasonintju**
## Description

- refactor AddressDisplay component
- refactor SwitchBetweenMantaSignerAndMantaWallet
- close #902
- close #907 
- update some styles
- remove manta signer download link

---

Before we can merge this PR, please make sure that all the following items have been checked off. If any of the checklist items are not applicable, please leave them but write a little note why.

- [ ] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work
- [ ] Re-reviewed files changed in the Github PR explorer

## If PR to `staging`
- [ ] Updated relevant documentation in the code
- [x] Test code changes manually, and describe your procedure in a comment on this PR
- [x] Mark this PR with the correct milestone

## If PR to `main`
- [ ] Update the version number in `package.json` and `src/config/common.json`
- [ ] Update the minimum required signer version number in `src/config/common.json` if applicable
- [ ] Test following the procedure in `docs/manual_test.md`
- [ ] Update `CHANGELOG.md`
- [ ] Make sure that all PR's to `manta-signer` and `sdk` on which this PR depends are merged

### CODE REVIEW

File: src/components/Accounts/AccountSelectDropdown.tsx [hunk 1]
```diff
@@ -10,6 +10,37 @@ import makeBlockie from 'ethereum-blockies-base64';
...
 
+type AddressDisplayProperties = {
+  isPrivateAddress?: boolean;
+  zkAddress: string,
+  publicAddress: string,
+  className?: string;
+  disabled: boolean;
+}
+
+const AddressDisplay = ({ isPrivateAddress, zkAddress, publicAddress, className, disabled }: AddressDisplayProperties) => (
```
Comment (jincan39) [inline src/components/Accounts/AccountSelectDropdown.tsx:21]: I didn't where is the `isPrivateAddress` logic defined?

File: src/components/Accounts/AccountSelectMenu.tsx [hunk 2]
```diff
@@ -17,10 +18,17 @@ const DisplayAccountsButton = () => {
...
+  const { usingMantaWallet } = useGlobal();
 
   const isMetamaskEnabled =
     !!ethAddress && window?.location?.pathname?.includes('bridge');
 
+  // using manta wallet zkAddress combined with other wallets public address
+  const isOnlyUsingMantaWalletZKAddress = 
+    usingMantaWallet 
+    && selectedWallet?.extensionName !== WALLET_NAME.MANTA 
+    && window?.location?.pathname?.includes('transact');
```
Comment (jincan39) [inline src/components/Accounts/AccountSelectMenu.tsx:30]: suggest to use useLocation and match the exact route to transact page

File: src/pages/SendPage/SwitchMantaWalletAndSigner.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    if (disabled) {
+      return;
+    }
+    const lastAccessExtensionName = getLastAccessedWallet()?.extensionName;
+    if (usingMantaWallet && lastAccessExtensionName === WALLET_NAME.MANTA) {
+      const substrateWallets = getSubstrateWallets();
+      const enabledExtentions = substrateWallets.filter((wallet) => (wallet.extension && wallet.extensionName !== WALLET_NAME.MANTA));
+      if (enabledExtentions.length > 0) {
+        // switch to another wallet as the default wallet
+        if (keyringIsBusy.current === false && !disabled) {
```
Comment (jincan39) [inline src/pages/SendPage/SwitchMantaWalletAndSigner.tsx:33]: `!disabled` is not necessary, line 24 has returned.

**Review by jincan39 — CHANGES_REQUESTED** [review]

File: src/components/Accounts/AccountSelectDropdown.tsx (hunk 1 above)
Comment (jasonintju) [reply src/components/Accounts/AccountSelectDropdown.tsx:21]: just a prop

File: src/components/Accounts/AccountSelectMenu.tsx (hunk 2 above)
Comment (jasonintju) [reply src/components/Accounts/AccountSelectMenu.tsx:30]: `transact` is enough. If we use exact route, we will have to check `calamari` and `dolphin`

File: src/pages/SendPage/SwitchMantaWalletAndSigner.tsx (hunk 3 above)
Comment (jasonintju) [reply src/pages/SendPage/SwitchMantaWalletAndSigner.tsx:33]: yeah, great catch

**Review by jincan39 — APPROVED** [review]

**Review by Kevingislason — APPROVED** [review]

**Pull request merged**